src/services/event.js

```diff
@@ -273,11 +277,83 @@ const makeMatchesFilters = (filters) => (event) => {
   return false;
 };
 
+/**
+ * Splits up multiple occurrences of events into individual events
+ * and re-sorts the array
+ *
+ * @param {Object[]} events the array of events to process
+ * @returns {Object[]} the flattened array of events after splitting multiple occurrences
+*/
+const processMultipleOccurences = (events) => {
+
+  var splitEvents = [];
+
+  for (let i = 0; i < events.length; i++) {
+
+    let thisEvent = events[i];
+
+    // if the event has more than 1 occurrences, loop over them and create new events for each; otherwise add the event itself
+    if (thisEvent.Occurrences && thisEvent.Occurrences.length > 1) {
+
+      for (let j = 0; thisEvent.Occurrences && j < thisEvent.Occurrences.length; j++) {
+
+        const thisOccurrence = thisEvent.Occurrences[j];
+        const thisOccurrenceStartDateTime = DateTime.fromISO(thisOccurrence.StartDate).toJSDate();
+        const thisOccurrenceEndDateTime = DateTime.fromISO(thisOccurrence.EndDate).toJSDate();
+        const thisOccurrenceDate = moment(thisOccurrenceStartDateTime).format('MMMM D, YYYY');
+        const thisOccurrenceTimeRange = ("" + moment(thisOccurrenceStartDateTime).format('h:mm A') + " - " + moment(thisOccurrenceEndDateTime).format('h:mm A'));
+        const allOccurrences = thisEvent.Occurrences.map((occurrence) => {return {start: occurrence.StartDate,
+          end: occurrence.EndDate}});
+
+        var thisEventOccurrence = {
+          Description: thisEvent.Description,
+          Event_ID: ("" + thisEvent.Event_ID + "-" + j),
+          Event_Name: thisEvent.Event_Name,
+          Event_Title: thisEvent.Event_Title,
+          Event_Type_Name: thisEvent.Event_Type_Name,
+          HasCLAWCredit: thisEvent.HasCLAWCredit,
+          IsPublic: thisEvent.IsPublic,
+          Organization: thisEvent.Organization,
+          date: thisOccurrenceDate,
+          location: (thisOccurrence.location ? thisOccurrence.location : thisEvent.location),
+          timeRange: thisOccurrenceTimeRange,
+          title: thisEvent.title,
+          startDateTime: thisOccurrenceStartDateTime,
+          allOccurrences: allOccurrences,
+          recurring: true
+        };
+
+        splitEvents.push(thisEventOccurrence);
```

Comment:
You're still using `var` here. Can we change this to `const`? Also, technically the intermediate variable isn't necessary, since you're only pushing it onto `splitEvents`. You could just push the object literal.

Also, this could be made a lot more concise using the spread operator:
```suggestion
        splitEvents.push( {
          ...thisEvent,
          Event_ID: ("" + thisEvent.Event_ID + "-" + j),
          date: thisOccurrenceDate,
          location: (thisOccurrence.location || thisEvent.location),
          timeRange: thisOccurrenceTimeRange,
          startDateTime: thisOccurrenceStartDateTime,
          allOccurrences: allOccurrences,
          recurring: true
        };);
```